### PR TITLE
Change the message for cut/delete and message of bottom-left frame

### PR DESCRIPTION
**Pull request by shapon2004**
https://apromore.atlassian.net/browse/AP-5276
https://apromore.atlassian.net/browse/AP-5272

### CODE REVIEW

File: Apromore-Clients/manager-client/src/main/java/org/apromore/manager/client/ManagerServiceImpl.java [hunk 1]
```diff
@@ -810,4 +810,39 @@ public void deleteElements(Map<SummaryType, List<VersionSummaryType>> elements,
...
+			boolean processPermission = procSrv.hasWritePermissionOnProcess(secSrv.getUserByName(username), processIds);
+			if (!processPermission)
+				return false;
+		}
+		if (!logIds.isEmpty()) {
+			boolean logPermission = logSrv.hasWritePermissionOnLog(secSrv.getUserByName(username), logIds);
+			if (!logPermission)
+				return false;
+		}
+		return true;
```
Comment (raboczi) [inline Apromore-Clients/manager-client/src/main/java/org/apromore/manager/client/ManagerServiceImpl.java]: Style-wise, it'd be preferable to include { } around the conditional bodies.  However my thought here would be to return a single expression, something like:
```
User user = secSrv.getUserByName(username);

return (folderIds.isEmpty()  || workspaceSrv.hasWritePermissionOnFolder(user, folderIds)
    && (processIds.isEmpty() || procSrv.hasWritePermissionOnProcess(user, processIds)
    && (logIds.isEmpty()     || logSrv.hasWritePermissionOnLog(user, logIds));
```
You could even go further and remove the `.isEmpty()` checks, which as far as I can tell aren't required -- you always have permission on an empty list of artifacts, and the service calls don't currently incur network overhead.

File: Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/mapper/WorkspaceMapper.java [hunk 2]
```diff
@@ -75,7 +77,8 @@
      */
     public static List<FolderType> convertFoldersToFolderTypes(List<GroupFolder> folders) {
         List<FolderType> folderTypes = new ArrayList<>();
-
+        DateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss");
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/mapper/WorkspaceMapper.java]: Better to use the constant at `org.apromore.common.Constants.DATE_FORMAT`.

File: Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/mapper/WorkspaceMapper.java [hunk 3]
```diff
@@ -86,6 +89,17 @@
...
             folder.setHasOwnership(node.isHasOwnership());
+			try {
+				if (node.getFolder().getCreatedBy() != null) {
+					folder.setOwnerName(node.getFolder().getCreatedBy().getUsername());
+				}
+
+				if (node.getFolder().getDateModified() != null)
+					folder.setLastUpdate(dateFormat.format(node.getFolder().getDateModified()));
+				else
+					folder.setLastUpdate(dateFormat.format(node.getFolder().getDateCreated()));
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/mapper/WorkspaceMapper.java]: { } around conditional bodies

File: Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/mapper/WorkspaceMapper.java [hunk 4]
```diff
@@ -86,6 +89,17 @@
...
+			} catch (Exception ex) {}
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/mapper/WorkspaceMapper.java]: Handle this exception, if only by logging it.

File: Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/ProcessServiceImpl.java [hunk 5]
```diff
@@ -842,4 +842,15 @@ private boolean isRequestForNativeFormat(Integer processId, String branch, Versi
...
 
+
+	@Override
+	public boolean hasWritePermissionOnProcess(User user, List<Integer> processIds) {
+		for (Integer processId : processIds) {
+			if (!canUserWriteProcess(user, processId)) {
+				return false;
+			}
+		}
+		return true;
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/ProcessServiceImpl.java]: Perhaps a stream version?
```
return processIds.stream().allMatch(processId -> canUserWriteProcess(user, processId));
```

File: Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/WorkspaceServiceImpl.java [hunk 6]
```diff
@@ -1053,4 +1053,14 @@ private void removeGroupUsermetadata(Group group, Usermetadata usermetadata) {
...
   }
 
+  @Override
+  public boolean hasWritePermissionOnFolder(User user, List<Integer> selectedFolders) {
+	for (Integer folderId : selectedFolders) {
+		if (!canUserWriteFolder(user, folderId)) {
+			return false;
+		}
+	}
+	return true;
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/WorkspaceServiceImpl.java]: Could be done with a stream:
```
return selectedFolders.stream().allMatch(folderId -> canUserWriteFolder(user, folderId));
```

File: Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/EventLogServiceImpl.java [hunk 7]
```diff
@@ -676,4 +676,18 @@ public Usermetadata savePerspectiveByLog(List<String> perspectives, Integer logI
...
+		for (Integer logId : logIds) {
+			try {
+				if (!canUserWriteLog(user.getUsername(), logId)) {
+					return false;
+				}
+			} catch (UserNotFoundException e) {
+				return false;
+			}
+		}
+		return true;
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/EventLogServiceImpl.java]: This could be a bit more concise:
```
try {
    String username = user.getUsername();

    return logIds.stream().allMatch(logId -> canUserWriteLog(username, logId));

} catch (UserNotFoundException e) {
    return false;
}
```

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/renderer/SummaryItemRenderer.java [hunk 8]
```diff
@@ -258,16 +270,19 @@ private void renderFolder(final Listitem listitem, final FolderType folder, fina
...
+        listitem.appendChild(new Listcell(""));
+        listitem.appendChild(new Listcell(""));
+        listitem.appendChild(renderFolderDate(folder.getLastUpdate()));
+        listitem.appendChild(renderFolderOwner(folder.getOwnerName()));
         // Append columns for any folder attributes supplied via plugins
         for (PortalProcessAttributePlugin plugin: plugins) {
             listitem.appendChild(plugin.getListcell(folder));
         }
-
+
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/renderer/SummaryItemRenderer.java]: Introduced trailing whitespace.

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/workspaceOptions/RenameFolderController.java [hunk 9]
```diff
@@ -50,21 +53,33 @@
...
+  private String folderName;
   private Logger LOGGER = PortalLoggerFactory.getLogger(AddFolderController.class);
 
   public RenameFolderController(MainController mainController, int folderId, String name)
       throws DialogException {
     this.mainController = mainController;
 
     try {
+    		
+
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/workspaceOptions/RenameFolderController.java]: Why is this whitespace introduced?

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/workspaceOptions/RenameFolderController.java [hunk 10]
```diff
@@ -50,21 +53,33 @@
...
+      this.folderName=name;
       this.btnSave = (Button) this.folderEditWindow.getFellow("btnSave");
       this.btnCancel = (Button) this.folderEditWindow.getFellow("btnCancel");
       this.folderId = folderId;
+      if(!this.mainController.getWorkspaceService().hasWritePermissionOnFolder(mainController.getSecurityService().getUserByName(UserSessionManager.getCurrentUser().getUsername()), Arrays.asList(this.folderId)))
+      {
+    	  Notification.error(Labels.getLabel("portal_noPrivilegeRename_message"));
+    	  if(this.folderEditWindow!=null)
+    	  this.folderEditWindow.detach();
+		  return;
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/workspaceOptions/RenameFolderController.java]: Add { } around conditional body.

File: Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/workspaceOptions/RenameFolderController.java [hunk 11]
```diff
@@ -87,13 +102,23 @@ public void onEvent(Event event) throws Exception {
           cancel();
         }
       });
+      Button resetB = (Button) this.folderEditWindow.getFellow("resetButton");
+      resetB.addEventListener("onClick", new EventListener<Event>() {
+        public void onEvent(Event event) throws Exception {
+        	resetFolderName();
+        }
+      });
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/workspaceOptions/RenameFolderController.java]: More concise with a lambda:
```
resetB.addEventListener("onClick", event -> {
    resetFolderName();
});
```

File: Apromore-Core-Components/Apromore-Portal/src/main/resources/zk-label.properties [hunk 12]
```diff
@@ -294,11 +295,11 @@ portal_unableSave_type = ERROR
 portal_langChanged_message = You need to reload the portal to see the effect
 portal_langChanged_title = Change language
 portal_langChanged_type = WARNING
-portal_itemsSelectedToMove_message = {0} item(s) have been selected to be moved
+portal_itemsSelectedToMove_message = {0} {0, choice, 1#item|1<items } have been selected to be moved
```
Comment (raboczi) [inline Apromore-Core-Components/Apromore-Portal/src/main/resources/zk-label.properties:298]: That's slick; I've haven't seen this done before.  👌

File: Apromore-Custom-Plugins/File-Portal-Plugin/src/main/java/org/apromore/plugin/portal/file/impl/EditMetadataController.java [hunk 13]
```diff
@@ -85,6 +88,13 @@ public EditMetadataController(MainController mainC,
 
     this.window = createComponent("zul/editlogdata.zul");
     // this.window.setTitle("Edit log metadata");
+    if(!this.mainController.getEventLogService().hasWritePermissionOnLog(mainController.getSecurityService().getUserByName(UserSessionManager.getCurrentUser().getUsername()), Arrays.asList(this.log.getId())))
+    {
+  	  Notification.error(Labels.getLabel("portal_noPrivilegeRename_message"));
+  	  if(this.window!=null)
+  	  this.window.detach();
```
Comment (raboczi) [inline Apromore-Custom-Plugins/File-Portal-Plugin/src/main/java/org/apromore/plugin/portal/file/impl/EditMetadataController.java]: { } around conditional body

**Review by raboczi — CHANGES_REQUESTED** [review]
Ideally, we'd have one PR per Jira issue.  I have several suggestions at a syntax level for consideration.

File: Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/EventLogServiceImpl.java (hunk 7 above)
Comment (shapon2004) [reply Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/EventLogServiceImpl.java]: Correction here.

File: Apromore-Core-Components/Apromore-Portal/src/main/resources/zk-label.properties (hunk 12 above)
Comment (shapon2004) [reply Apromore-Core-Components/Apromore-Portal/src/main/resources/zk-label.properties:298]: I have found it easy to handle conditional case.

**Review by raboczi — APPROVED** [review]
